python/delta/tests/test_deltatable.py

```diff
@@ -850,6 +850,54 @@ def runRestoreToVersion() -> None:
         self.__intercept(runRestoreToVersion,
                          "version needs to be an int but got '<class 'str'>'")
 
+    def test_optimize(self) -> None:
```

Comment:
can you also add a negative test which provides non-partition column filter?

Reply:
@vkorukanti I added tests for non-partition columns.

I also renamed ``partitionFilter`` method as ``where`` as you pointed [here](https://github.com/delta-io/delta/pull/961#discussion_r861114399).